### PR TITLE
[3/3] Refactor response status in survey to chips for easier differentiation and change unsubmit to icon

### LINKED ISSUE #4602: Add chips to differentiate between various response statuses for survey
Related to #4599 #1914 

https://coursemology.org/courses/course-id/surveys/survey-id/responses

In survey responses, add chips to differentiate between various response statuses

### PR DESCRIPTION
**Pull request by leeroy999**
Fixes #4602

- https://github.com/Coursemology/coursemology2/pull/4599
- https://github.com/Coursemology/coursemology2/pull/4622

Changes:
- Update status links to chips
- Update unsubmit button to icon
- Refactor translations in survey index to a new file

### CODE REVIEW

**Comment by cysjonathan** [comment]
@leeroy999 the entire chip should be clickable (act as a link/button), not just the text inside.

File: client/app/bundles/course/survey/pages/ResponseIndex/index.jsx [hunk 1]
```diff
@@ -159,6 +159,15 @@ const ResponseIndex = (props) => {
     return submittedAt;
   };
 
+  const handleUnsubmitResponse = (buttonId) => {
```
Comment (ekowidianto) [inline client/app/bundles/course/survey/pages/ResponseIndex/index.jsx:162]: rename buttonId to responseId

**Review by ekowidianto — CHANGES_REQUESTED** [review]
Thanks! See comments

File: client/app/bundles/course/survey/constants.js [hunk 2]
```diff
@@ -96,7 +96,7 @@ const actionTypes = mirrorCreator([
 export const workflowStates = {
   Unstarted: 'unstarted',
   Attempting: 'attempting',
-  Submitted: 'submitted',
+  Published: 'published',
```
Comment (ekowidianto) [inline client/app/bundles/course/survey/constants.js:99]: can I check why was this changed to published?

File: client/app/lib/components/UnsubmitButton.jsx [hunk 3]
```diff
@@ -19,12 +19,7 @@ class UnsubmitButton extends Component {
           <IconButton
             id={`unsubmit-button-${this.props.buttonId}`}
             disabled={isUnsubmitting}
-            onClick={() =>
-              this.props.setState({
-                ...this.props.state,
-                unsubmitConfirmation: true,
-              })
-            }
+            onClick={this.props.onClick}
```
Comment (ekowidianto) [inline client/app/lib/components/UnsubmitButton.jsx:22]: Can we re-write this as a functional component?

File: client/app/lib/components/UnsubmitButton.jsx [hunk 4]
```diff
@@ -42,8 +37,7 @@ UnsubmitButton.propTypes = {
   buttonId: PropTypes.number.isRequired,
   color: PropTypes.string.isRequired,
   isUnsubmitting: PropTypes.bool.isRequired,
-  setState: PropTypes.func.isRequired,
-  state: PropTypes.object.isRequired,
+  onClick: PropTypes.func.isRequired,
 };
 
 export default connect((state, ownProps) => ({
```
Comment (ekowidianto) [inline client/app/lib/components/UnsubmitButton.jsx:43]: Instead of getting the state directly from redux, can we pass the state down from the parent component instead?

File: client/app/lib/components/UnsubmitButton.jsx [hunk 5]
```diff
@@ -42,8 +37,7 @@ UnsubmitButton.propTypes = {
   buttonId: PropTypes.number.isRequired,
   color: PropTypes.string.isRequired,
   isUnsubmitting: PropTypes.bool.isRequired,
```
Comment (ekowidianto) [inline client/app/lib/components/UnsubmitButton.jsx:39]: Rename to disabled

**Review by ekowidianto — CHANGES_REQUESTED** [review]
See more comments

File: client/app/lib/components/UnsubmitButton.jsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  formButton: {
+    padding: '0.25em 0.4em',
+  },
+};
+
+class UnsubmitButton extends Component {
+  render() {
+    const { isUnsubmitting } = this.props;
+    return (
+      <>
```
Comment (ekowidianto) [inline client/app/lib/components/UnsubmitButton.jsx:17]: Empty JSX not needed

**Comment by ekowidianto** [comment]
Alright, I didnt realise the unsubmitButton (now SurveyUnsubmitButton) handles its own survey unsubmission logic. In this case, I dont think it is a good idea to share the unsubmit button component with assessment. Could you then revert the shared unsubmit button component - let assessment submission uses its own unsubmit button? For survey unsubmission, you can refactor the current SurveyUnsubmitButton to allow you to choose either to use button with unsubmit text or icon button with tooltip.

Remember to only create 1 ReactTooltip component to be shared across different icon button.

File: client/app/bundles/course/survey/constants.js (hunk 2 above)
Comment (leeroy999) [reply client/app/bundles/course/survey/constants.js:99]: The submitted state is grey color for assessment as its not the final stage, but submitted is the final stage for surveys, so I changed it to have the green color:

Previous: image
Current: image

Or is there a better way of doing this?

File: client/app/bundles/course/survey/pages/ResponseIndex/index.jsx [hunk 7]
```diff
@@ -192,7 +210,24 @@ const ResponseIndex = (props) => {
...
               {response.status === responseStatus.SUBMITTED &&
               response.canUnsubmit ? (
-                <UnsubmitButton responseId={response.id} />
+                <>
+                  <UnsubmitButton
+                    buttonId={response.id}
+                    color={palette.submissionIcon.unsubmit}
+                    onClick={onClick}
+                  />
+                  <ReactTooltip id="unsubmit-button" effect="solid">
```
[the remark this answers is not in the document]
Comment (cysjonathan) [reply client/app/bundles/course/survey/pages/ResponseIndex/index.jsx:219]: Sounds familiar 👍

File: client/app/bundles/course/survey/constants.js (hunk 2 above)
Comment (cysjonathan) [reply client/app/bundles/course/survey/constants.js:99]: I think it is okay to leave the gray submitted colour consistent for both submissions as well as assessment.
No need to change.
